lib/util.js

```diff
@@ -352,5 +352,27 @@ exports.stripHTML = html => html
   .replace(/[\n\r]/g, ' ')
   .replace(/\s*<\s*br\s*\/?\s*>\s*/gi, '\n')
   .replace(/<\s*\/\s*p\s*>\s*<\s*p[^>]*>/gi, '\n')
+  .replace(/<a\s+(?:[^>]*?\s+)?href=(?:["'])\/redirect.*?q=(.*?)(?:[&'"]).*?<\/a>/gi, '$1')
+  .replace(/<a\s+(?:[^>]*?\s+)?href=(?:["'])(https?.*?)(?:['"]).*?<\/a>/gi, '$1')
+  .replace(/<a\s+(?:[^>]*?\s+)?href=(?:["'])((?:\/results?|\/watch?).*?)(?:['"]).*?<\/a>/gi, 'https://youtube.com$1')
   .replace(/<.*?>/gi, '')
   .trim();
+
+/**
+ * Get a number from an abbreviated number string.
+ *
+ * @param {string} string
+ * @returns {number}
+ */
+exports.parseAbbreviatedNumber = string => {
+  const match = string
+    .replace(',', '.')
+    .replace(' ', '')
+    .match(/([\d,.]+)([MK]?)/);
+  if (match) {
+    const [, num, multi] = match;
+    return multi === 'M' ? +num * 1000000 :
+      multi === 'K' ? +num * 1000 : +num;
+  }
+  return null;
+};
```

Comment:
The abbreviated number now gets matched once, and number and multiplier are extracted from the second and third match group.

Also, if it doesn't match the regex, the function returns null. The isNaN check is therefore not necessary anymore.